include/sparrow/array/array_data_factory.hpp

```diff
@@ -126,7 +127,7 @@ namespace sparrow
      * @param offset The offset of the array data.
      * @return The created array_data object.
      */
-    template <constant_range_for_array_data ValueRange>
+    template <range_for_array_data ValueRange>
```

Comment:
Just a note: looking at this concept and it's constant version and trying to understand what's wrong lead me to think `constant_range` and `constant_iterator` are not correct. If you are not already on it I'll take note to investigate this.

Reply:
I did not yet tried to fix it, but they indeed are not working as expected, at least for std::vector<bool>

But I believe the restriction to a const_range is un-needed anyhow

Reply:
My suspicion is that these concepts should be requiring that `std::ranges::cbegin/cend` works, instead of `std::ranges::begin/end` although I didnt test this theory yet.